File: app/src/main/java/com/adisa/diningplus/db/dao/MealDao.java
package com.adisa.diningplus.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.adisa.diningplus.db.entities.Meal;

import java.util.List;

@Dao
public interface MealDao {

    @Query("SELECT * FROM meals")
    List<Meal> getAll();

    @Query("SELECT * FROM meals WHERE hall_id = :hallId")
    List<Meal> getHall(String hallId);

    @Insert
    void insert(Meal meal);

    @Delete
    void delete(Meal meal);

    @Update
    void update(Meal meal);

    @Query("DELETE FROM meals WHERE hall_id = :hallId")
    void clearHall(String hallId);
}

File: app/src/main/java/com/adisa/diningplus/db/dao/MealDao.kt
package com.adisa.diningplus.db.dao

import androidx.room.*
import com.adisa.diningplus.db.entities.Meal

@Dao
interface MealDao {
    @Query("SELECT * FROM meals WHERE hall_id = :hallId")
    fun getHall(hallId: String): List<Meal>

    @Insert
    fun insert(meal: Meal)

    @Delete
    fun delete(meal: Meal)

    @Update
    fun update(meal: Meal)

    @Query("DELETE FROM meals WHERE hall_id = :hallId")
    fun clearHall(hallId: String)
}